src/messaging/ExchangeMgr.cpp

```diff
@@ -161,17 +174,42 @@ ExchangeContext * ExchangeManager::AllocContext(uint16_t ExchangeId, uint64_t Pe
 void ExchangeManager::DispatchMessage(const PacketHeader & packetHeader, const PayloadHeader & payloadHeader,
                                       System::PacketBufferHandle msgBuf)
 {
+    CHIP_ERROR err                          = CHIP_NO_ERROR;
     UnsolicitedMessageHandler * umh         = nullptr;
     UnsolicitedMessageHandler * matchingUMH = nullptr;
-    CHIP_ERROR err                          = CHIP_NO_ERROR;
+    ExchangeContext * ec                    = nullptr;
+    const uint16_t protocolId               = payloadHeader.GetProtocolID();
+    const uint8_t messageType               = payloadHeader.GetMessageType();
+    bool sendAckAndCloseExchange            = false;
+
+    // Received Delayed Delivery Message: Extend time for pending retrans objects
+    if (protocolId == Protocols::kProtocol_Protocol_Common && messageType == Protocols::Common::kMsgType_RMP_Delayed_Delivery)
+    {
+        const uint8_t * p              = msgBuf->Start();
+        const uint32_t PauseTimeMillis = LittleEndian::Read32(p);
+        const uint64_t DelayedNodeId   = LittleEndian::Read64(p);
```

Comment:
What guarantees there are enough bytes in the buffer?

Reply:
Added length check before decoding